### ISSUE TITLE
Optional parameter in / not working

### ISSUE THREAD

**Issue by pupaxxo** [dannyvankooten/AltoRouter#98]
I tried to match this url '/page/[i:page]?' and it works like a charms, but when i try '/[i:page]?' it doesn't work why?

**Comment by rakshazi** [comment]
@koenpunt @dannyvankooten hello!
I have the same problem, but with another routes: example.com/<int>-<string|int>-<string|int>-<...>-<string|int>
example route: example.com/371-kak-umnozhat-lyube-chisla-v-ume-vse7-444-podzabtj-shkolnj-sposob
I got regex from comments above and modified it, but it didn't help. Here is it: `'@^/(?P<slug>[0-9A-Za-z]++)?$'`

Please, explain, how can I fix this?

Best regards, 
Nikita.

**Comment by rakshazi** [comment]
Oh, found needed config myself, here is it:
```php
$router->addMatchTypes(array('post_slug' => '[0-9A-Za-z_-]++'));
```
And.. it works :)
Thanks!

**Comment by mastacontrola** [comment]
Updating here so all can follow as well. Pull request #99 should address the problem.

I know the code's not merged or anything but at least it shows a potential fix for the problem directly in the core code.

**Comment by koenpunt** [comment]
This is now resolved since #99 got merged
